### PR TITLE
Add api test for user authentication and management

### PR DESCRIPTION
**Pull request by wangcan0329**

### CODE REVIEW

File: rest-server/test/userToken.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      .set('Authorization', 'Bearer ' + validToken)
+      .send(JSON.parse(global.mustache.render(getTokenTemplate, { 'username': 'new_user', 'password': 'abcdef' })))
+      .end((err, res) => {
+        global.chai.expect(res, 'status code').to.have.status(401);
+        global.chai.expect(res, 'response format').be.json;
+        global.chai.expect(res.body.message, 'response message').equal('authentication failed');
+        done();
+      });
+  });
+
```
Comment (hwuu) [inline rest-server/test/userToken.js]: Add another negative case: user name does not exist.

File: rest-server/test/userManager.js [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+          'modifiedIndex': 23,
+          'createdIndex': 23
+        }
+      });
+  });
+
+  //
+  // Negative cases
+  //
+
```
Comment (hwuu) [inline rest-server/test/userManager.js]: Add a positive case here?

File: rest-server/test/userManager.js [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+      .set('Authorization', 'Bearer ' + validToken)
+      .send(JSON.parse(global.mustache.render(newUserTemplate, { 'username': 'new_user', 'password': 'abcdef', 'admin': false, 'modify': true })))
+      .end((err, res) => {
+        global.chai.expect(res, 'status code').to.have.status(201);
+        global.chai.expect(res, 'response format').be.json;
+        global.chai.expect(res.body.message, 'response message').equal('update successfully');
+        done();
+      });
+  });
+
```
Comment (hwuu) [inline rest-server/test/userManager.js]: Add negative cases? Update a non-existing user, etc. And what if password is set to an empty string?

File: rest-server/test/userManager.js [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+        'prevNode': {
+          'key': '/users/new_user/passwd',
+          'value': '8507b5d862306d5bdad95b3d611b905ecdd047b0165ca3905db0d861e76bce8f3a8046e64379e81f54865f7c41b47e57cec887e5912062211bc9010afea8ab12',
+          'modifiedIndex': 10,
+          'createdIndex': 10
+        }
+      });
+
+      nock(etcdHosts)
+      .put('/v2/keys/users/new_user/admin', { 'value': 'false' })
```
Comment (hwuu) [inline rest-server/test/userManager.js]: Indent 2 spaces.

File: rest-server/test/userManager.js [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+    global.chai.request(global.server)
+      .put('/api/v1/user')
+      .set('Authorization', 'Bearer ' + validToken)
+      .send(JSON.parse(global.mustache.render(newUserTemplate, { 'username': 'new_user', 'password': '123456', 'admin': true, 'modify': true })))
+      .end((err, res) => {
+        global.chai.expect(res, 'status code').to.have.status(500);
+        global.chai.expect(res, 'response format').be.json;
+        global.chai.expect(res.body.message, 'response message').equal('update failed');
+        done();
+      });
```
Comment (hwuu) [inline rest-server/test/userManager.js]: Add another negative case: what if the user that calls the api is not an admin?

File: rest-server/test/userToken.js (hunk 1 above)
Comment (wangcan0329) [reply rest-server/test/userToken.js]: Sure

File: rest-server/test/userManager.js (hunk 2 above)
Comment (wangcan0329) [reply rest-server/test/userManager.js]: There is a positive case behind it

File: rest-server/test/userManager.js (hunk 3 above)
Comment (wangcan0329) [reply rest-server/test/userManager.js]: if password sets to an empty string will trigger parameter validation error. "password\" is not allowed to be empty.
Already add two negative case:1) update a non-existing user 2)update password to be empty

File: rest-server/test/userManager.js (hunk 5 above)
Comment (wangcan0329) [reply rest-server/test/userManager.js]: Add negative case to add, update and delete user with non-admin token.

**Pull request merged**